### ISSUE TITLE
Error: unexpected token tCOLON on valid code

### ISSUE THREAD

**Issue by Zyphrax** [bbatsov/rubocop#1889]
I have the following Ruby code:

```ruby
LOCALES = {
  'en-US': 'English / United States',
  'en-UK': 'English / United Kingdom',
  nl: 'Dutch / Netherlands'
}
```
When I run Rubocop:
```
Offenses:

config/initializers/app_settings.rb:9:10: E: unexpected token tCOLON
  'en-US': 'English / United States',
```

As far as I know `'string1': 'string2'` is a valid hash notation in > ruby 1.9.
Ruby / Rails doesn't seem to complain when I run it. Is this a small Rubocop bug or is it me?

Rubocop version 0.31.0

**Comment by bbatsov** [comment]
Tested this in Ruby 2.1 and it results in a syntax error there. Are you sure this is valid syntax?

**Comment by Zyphrax** [comment]
No syntax error here:
```
Loading development environment (Rails 4.2.1)
irb(main):001:0> a = { 'en-US': 'English' }
=> {:"en-US"=>"English"}
```

I'm running ruby 2.2.2p95 (2015-04-13 revision 50295) [x86_64-linux-gnu]

**Comment by bbatsov** [comment]
Seems this was changed in Ruby 2.2. //cc @whitequark

**Comment by Zyphrax** [comment]
This is quite insightful. See point 1 of mu is too short. It seems to be a change in Ruby 2.2.
http://stackoverflow.com/questions/10004158/is-hash-rocket-deprecated

I work under OSX but my VM with Ubuntu runs the code. So probably my Rubocop plugin is detecting this as a syntax error (because of Ruby 2.1), but it is running fine in Ubuntu (because of Ruby 2.2).

**Comment by Zyphrax** [comment]
Funny thing is that Rubocopy changed it to the '...': '....' notation with the --auto-correct option.
Perhaps the auto-correct should leave string hashes alone for now.

**Comment by whitequark** [comment]
Hrm.
```
$ ./bin/ruby-parse --22 -e '{"x":y}'
(hash
  (pair
    (sym :x)
    (send nil :y)))
```
